File: src/main/java/it/unicam/cs/diciottoPolitico/casotto/entity/OrdinazioneBar.java
package it.unicam.cs.diciottoPolitico.casotto.entity;


/**
 * Rappresenta un'ordinazione bar.
 *
 * @see ArticoloBar
 */
public interface OrdinazioneBar extends Vendita{

    /**
     * Ritorna l' {@link ArticoloBar} associato all'ordinazione.
     *
     * @return l'articolo bar associato all'ordinazione
     * @see ArticoloBar
     */
    ArticoloBar getArticoloBar();

    /**
     * Ritorna lo stato dell'ordinazione bar.
     *
     * @return lo stato dell'ordinazione bar
     */
    StatusOrdinazioneBar getStatusOrdinazioneBar();

    /**
     * Imposta lo status dell'ordinazione bar.
     *
     * @param statusOrdinazioneBar il nuovo status dell'ordinazione bar
     * @throws NullPointerException se lo status &egrave; nullo
     */
    void setStatusOrdinazioneBar(StatusOrdinazioneBar statusOrdinazioneBar);

}

File: Casotto-Front-End/src/app/model/ordinazione-bar.ts
import { ArticoloBar } from "./articolo-bar";
import { StatusOrdinazioneBar } from "./status-ordinazione";
import { Vendita } from "./vendita";

export interface OrdinazioneBar {
    id: string;
    articoloBar: ArticoloBar;
    statusOrdinazioneBar: StatusOrdinazioneBar;
    vendita: Vendita;
    codiceSpiaggia: string;
}